test/controllers/pledges_controller_test.rb

```diff
@@ -1,89 +1,99 @@
-#       assert_equal @pledge.history_tracks.count, 2
-#       @changes = @pledge.history_tracks.last
-#       assert_equal @changes.modified[:updated_by_id], @user.id
-#       assert_equal @changes.modified[:pledge_type], 'Edited Pledge'
-#     end
-
-#     it 'should refuse to save pledge type to blank' do
-#       [nil, ''].each do |bad_text|
-#         assert_no_difference 'Pregnancy.find(@pregnancy).pledges.find(@pledge).history_tracks.count' do
-#           @pledge_edits[:pledge_type] = bad_text
-#           patch :update, pregnancy_id: @pregnancy, id: @pledge, pledge: @pledge_edits
-#           assert_response :bad_request
-#           @pledge.reload
-#           assert_equal @pledge.pledge_type, 'Edited Pledge'
-#         end
-#       end
-#     end
-#   end
-
-# end
+
+  before do
+    @user = create :user
+    sign_in @user
+    @patient = create :patient
+    @pregnancy = create :pregnancy, patient: @patient
+  end
+
+  describe 'create method' do
+    before do
+      @pledge = attributes_for :pledge
+
+#	NOTE: :pledge is getting passed in as a string			
+# 			@pledge = create :pledge,
+# 								pledge_type: "Sample Pledge", 
+# 								amount: 5, 
+# 								other_pledge_identifier: "none", 
+# 								sent: DateTime.current, 
+# 								sent_by: @user, 
+# 								paid: false,
+# 								paid_date: DateTime.current
+
+      post :create, pregnancy_id: @pregnancy.id, pledge: @pledge
+    end
+
+    it 'should create and save a new pledge' do
+      assert_difference 'Pregnancy.find(@pregnancy).pledges.count', 1 do
+        post :create, pregnancy_id: @pregnancy.id, pledge: @pledge
+      end
+    end
+
+  # NOTE: it's redirecting so status code 302 (controller redirects to edit_pregnancy_path)		
+  #   it 'should respond success if the pledge submits' do
+  #     assert_response :success
+  #   end
+
+  # NOTE: template does not exist yet	and controller redirects to edit_pregnancy_path
+  #   it 'should render create.js.erb if it successfully saves' do
+  #     assert_template 'pledges/create'
+  #   end
+
+    it 'should redirect to edit pregnancy path if it saves' do
+      assert_redirected_to edit_pregnancy_path(@pregnancy)
+    end
+
+    it 'should log the creating user' do
+      assert_equal Pregnancy.find(@pregnancy).pledges.last.created_by, @user
+    end
+  end		
+
+  describe 'update method' do		
+    before do
+      @pledge = create :pledge, pregnancy: @pregnancy, pledge_type: 'Original Pledge'
+      @pledge_edits = {pledge: {pledge_type: 'Edited Pledge'}}
+      patch :update, pregnancy_id: @pregnancy, id: @pledge, pledge: @pledge_edits, format: :js
+      @pledge.reload
+    end
+
+#	NOTE: I created a blank template for this test	
+    it 'should render the correct template' do
+      assert_template 'pledges/update'
+    end
+
+    it 'should respond with success' do
+      assert_response :success
+    end
+
+  # NOTE: should work, but the call to reload is causing the factory attribute to reload, 
```

Comment:
INTERESTING. It sounds like there's something about the update method that's preventing it from actually saving -- we're running nearly identical test code for the notes controller and it seems to be working okay. 

Let me take a deeper look tomorrow.